pkg/runtime.v2/framework/core/framework.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2024 The Kubeflow Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	kubeflowv2 "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v2alpha1"
+	runtime "github.com/kubeflow/training-operator/pkg/runtime.v2"
+	"github.com/kubeflow/training-operator/pkg/runtime.v2/framework"
+	fwkplugins "github.com/kubeflow/training-operator/pkg/runtime.v2/framework/plugins"
+)
+
+type Framework struct {
+	registry                     fwkplugins.Registry
+	plugins                      map[string]framework.Plugin
+	enforceMLPlugins             []framework.EnforceMLPolicyPlugin
+	enforcePodGroupPolicyPlugins []framework.EnforcePodGroupPolicyPlugin
```

Comment:
Why do we have array of PodGroupPolicyPlugins ?

Reply:
Because we do not know if a single PodGroup mechanism is installed in the cluster.
Some TrainJob uses scheduler-plugins, and other TrainJob uses volcano could happen.

Reply:
Do you see a use-case when single cluster can have multiple Gang Scheduler plugins ?
Additionally, can we install Training Operator V2 if cluster doesn't have all plugins (e.g. doesn't know about PodGroup CRD) ?

Reply:
> Do you see a use-case when single cluster can have multiple Gang Scheduler plugins ?

IMO, there are no advantages. 

I'm wondering if we can add validations if registered `enforcePodGroupPolicyPlugins` is only one.
But, to realize the mechanism, we need to expose the plugin exposing mechanism and implement Configuration API as we discussed previously and in another thread.

So, now, there are no limitations.